src/Equinor.ProCoSys.IPO.Query/GetInvitations/GetInvitationsQueryBase.cs

```diff
@@ -0,0 +1,202 @@
+                            (filter.IpoStatuses.Contains(IpoStatus.Canceled) && invitation.Status == IpoStatus.Canceled)) &&
+                      (string.IsNullOrEmpty(filter.IpoIdStartsWith) ||
+                            invitation.Id.ToString().StartsWith(ipoIdStartWith)) &&
+                      (string.IsNullOrEmpty(filter.CommPkgNoStartsWith) ||
+                            invitation.CommPkgs.Any(c => c.CommPkgNo.ToUpper().StartsWith(filter.CommPkgNoStartsWith.ToUpper())) ||
+                            invitation.McPkgs.Any(mc => mc.CommPkgNo.ToUpper().StartsWith(filter.CommPkgNoStartsWith.ToUpper()))) &&
+                      (string.IsNullOrEmpty(filter.McPkgNoStartsWith) ||
+                            invitation.McPkgs.Any(mc => mc.McPkgNo.ToUpper().StartsWith(filter.McPkgNoStartsWith.ToUpper()))) &&
+                      (string.IsNullOrEmpty(filter.TitleStartsWith) ||
+                            invitation.Title.ToUpper().StartsWith(filter.TitleStartsWith.ToUpper())) &&
+                      (filter.PersonOid == null ||
+                            invitation.Participants.Any(p => p.AzureOid == filter.PersonOid)) &&
+                      (filter.FunctionalRoleCode == null ||
+                       invitation.Participants.Any(p => p.FunctionalRoleCode.ToUpper() == filter.FunctionalRoleCode.ToUpper())) &&
+                      (filter.PunchOutDateFromUtc == null ||
+                            invitation.StartTimeUtc >= filter.PunchOutDateFromUtc) &&
+                      (filter.PunchOutDateToUtc == null ||
+                            invitation.EndTimeUtc <= filter.PunchOutDateToUtc) &&
+                      (filter.LastChangedAtFromUtc == null ||
+                            (invitation.ModifiedAtUtc ?? invitation.CreatedAtUtc) >= filter.LastChangedAtFromUtc) &&
+                      (filter.LastChangedAtToUtc == null ||
+                            (invitation.ModifiedAtUtc ?? invitation.CreatedAtUtc) <= filter.LastChangedAtToUtc)
+                select new InvitationDto
+                {
+                    Id = EF.Property<int>(invitation, "Id"),
+                    ProjectName = invitation.ProjectName,
+                    Title = invitation.Title,
+                    Description = invitation.Description,
+                    Status = invitation.Status,
+                    Type = invitation.Type,
+                    CreatedAtUtc = invitation.CreatedAtUtc,
+                    StartTimeUtc = invitation.StartTimeUtc,
+                    EndTimeUtc = invitation.EndTimeUtc,
+                    CompletedAtUtc = invitation.CompletedAtUtc,
+                    AcceptedAtUtc = invitation.AcceptedAtUtc,
+                    ContractorRep = GetContractorRep(invitation.Participants.ToList()),
+                    ConstructionCompanyRep = GetConstructionCompanyRep(invitation.Participants.ToList()),
+                    McPkgNos = invitation.McPkgs.Select(mc => mc.McPkgNo).ToList(),
```

Comment:
I really don't understand how the McPkgs/CommPgs are fetched from DB here ... when seeing code, I was pretty sure that both list of McPkgs/CommPgs was empty after quering. But no, it works and I see that U cover it in tests.

Du you understand the magic here? Do .Select(mc => mc.McPkgNo) actually build a join in the executed query?

Reply:
I do not know the magic behind this, actually expected it to be empty myself, and that I needed to have .Include for all of these, but it works as is.